search/ranker.cpp

```diff
@@ -391,21 +445,22 @@ void Ranker::UpdateResults(bool lastUpdate)
 
     BailIfCancelled();
 
+    auto const & rankerResult = m_tentativeResults[i];
+
     if (m_params.m_viewportSearch)
     {
       // Viewport results don't need calculated address.
-      Result res = GenerateFinalResult(m_tentativeResults[i], false /* needAddress */);
+      Result res = MakeResult(rankerResult, false /* needAddress */, false /* needHighlighting */);
       m_emitter.AddResultNoChecks(move(res));
     }
     else
     {
       if (count >= m_params.m_limit)
         break;
 
-      LOG(LDEBUG, (m_tentativeResults[i]));
-
-      auto const & rankerResult = m_tentativeResults[i];
-      if (m_emitter.AddResult(MakeResult(rankerResult)))
+      LOG(LDEBUG, (rankerResult));
+      Result res = MakeResult(rankerResult, true /* needAddress */, true /* needHighlighting */);
```

Comment:
```c++
auto result = ...
```

Reply:
The explicit type here is to make it more clear that we're working with Result (and not with RankerResult or PreRankerResult).